bitcoin/src/crypto/ecdsa.rs

```diff
@@ -293,6 +295,16 @@ impl From<DecodeError> for ParseSignatureError {
     fn from(e: DecodeError) -> Self { Self::Decode(e) }
 }
 
+#[cfg(feature = "arbitrary")]
+impl<'a> Arbitrary<'a> for Signature {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(Signature{
+            signature: secp256k1::ecdsa::Signature::from_str(u.arbitrary()?).unwrap(),
```

Comment:
How does this work? An arbitrary string passed to a function that only succeeds if the string is a 144 character hex string. Or is there some magic that I'm missing?

Reply:
ah, i just realized this doesn't work. I saw it being used in the test below but didn't check. I'll have to figure out a way to create DER-encoded strings for this using the arbitrary bytes or else it'll throw errors

Reply:
updated to create `Signature` correctly